lib/1.4/utility.dml

```diff
@@ -330,6 +330,7 @@ template hard_reset {
         foreach obj in (this._each_hard_reset)
             obj.hard_reset();
     }
+    shared method hard_reset_deassert() default {}
```

Comment:
This is redundant in a sense; you can override signal_lower and call `default` instead.

(One could argue that it would be better with a simulator-agnostic method to override, but that sort of functionality would rather belong to the `signal_port` template.)

Reply:
Would be better to hide the implementation and expose only the high level methods as hard_reset()
is hreset;
port HRESET {
    saved bool reset_asserted;
    implement signal {
        method signal_lower() {
            hard_reset_deassert();
            default();
        }

Reply:
I'm with Erik on this. What's the added value of overriding `hard_reset_deassert` instead of overriding `signal_lower`?

If we add a `hard_reset_deassert` method like this we'd need to consider some more questions:
For example
* Should `hard_reset_deassert` be called after a `dev.hard_reset()`?
* Should `reset_asserted` be true during a call to `dev.hard_reset()`?
* Why is there no `hard_reset_assert()` method?

If we instead let the user override `signal_lower` in the HRESET port it's very clear what behavior is being overridden, and what can be expected.

Reply:
agreed